pkg/kubeapiserver/options/webhook_util.go

```diff
@@ -0,0 +1,64 @@
+package options
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"strings"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apiserver/pkg/util/webhook"
+	"k8s.io/klog/v2"
+)
+
+// newWebhookDialer return a DialFunc
+func newWebhookDialer(serviceResolver webhook.ServiceResolver, baseDialer utilnet.DialFunc) utilnet.DialFunc {
+	var d net.Dialer
+	delegateDialer := d.DialContext
+	if baseDialer != nil {
+		delegateDialer = baseDialer
+	}
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+		if serviceResolver == nil {
+			return delegateDialer(ctx, network, addr)
+		}
+		var (
+			host, port string
+			err        error
+		)
+		host, port, err = net.SplitHostPort(addr)
+		if err != nil {
+			klog.V(6).Infof("splitting hostport %s error: %v", addr, err)
+			// maybe just missing port
```

Comment:
you can assert the missing port error
```
		return "", -1, &net.AddrError{Err: "missing port in address", Addr: addr}
```

Reply:
I think and test again, we should not give a default port when missing port that is an internal network error, so we don't need to assert this